src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -295,17 +307,36 @@ public virtual VerifyResult VerifyStateDependent(StoreView snapshot, Transaction
                 if (!attribute.Verify(snapshot, this))
                     return VerifyResult.Invalid;
             long net_fee = NetworkFee - Size * NativeContract.Policy.GetFeePerByte(snapshot);
-            if (!this.VerifyWitnesses(snapshot, net_fee, WitnessFlag.StateDependent))
-                return VerifyResult.Invalid;
+
+            UInt160[] hashes = GetScriptHashesForVerifying(snapshot);
+            if (hashes.Length != witnesses.Length) return VerifyResult.Invalid;
+            for (int i = 0; i < hashes.Length; i++)
+            {
+                if (witnesses[i].VerificationScript.IsSignatureContract())
+                    net_fee -= SignatureContractCost;
```

Comment:
It doesn't need to consider the factor?

Reply:
It'll be added in another blocked pr https://github.com/neo-project/neo/pull/2045